demo-playground/BlockGeneration.hs

```diff
@@ -22,31 +24,38 @@ forkCoreNode :: forall m p c.
                 ( Mock.SimpleBlockCrypto c
                 , MonadSTM m
                 , MonadTimer m
-                , MonadFork m
                 , MonadIO m
-                , RunOuroboros p (DemoLedgerState (Mock.SimpleBlock p c))
+                , ProtocolLedgerView (Mock.SimpleBlock p c)
                 )
              => TVar m (DemoLedgerState (Mock.SimpleBlock p c))
-             -> OuroborosState p
+             -> OuroborosNodeConfig p
+             -> OuroborosNodeState p
              -> TVar m (Mempool Mock.Tx)
              -> TVar m (ChainProducerState (Mock.SimpleBlock p c))
              -> Duration (Time m)
              -> m ()
-forkCoreNode ledgerVar protocolState mempoolVar varCPS slotDuration = do
+forkCoreNode ledgerVar cfg initState mempoolVar varCPS slotDuration = do
     slotVar   <- atomically $ newTVar 1
     initDRG   <- liftIO getSystemDRG
 
-    let runProtocol :: MonadPseudoRandomT SystemDRG (OuroborosStateT p (Tr m)) a
+    let runProtocol :: forall a. MonadPseudoRandomT SystemDRG (OuroborosNodeStateT p (Tr m)) a
                     -> Tr m a
         runProtocol m =
-            fst <$> runOuroborosStateT (fst <$> withDRGT m initDRG) protocolState
+            -- TODO: This is not correct, this re-runs from the initial state
+            -- each time.
+            fst <$> runOuroborosNodeStateT (fst <$> withDRGT m initDRG) initState
```

Comment:
Yup, my bad -- I should have created a `TVar` for the random seed and update it as we did run the protocol 🤦‍♂️